flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/functions/BuiltInFunctionTestBase.java

```diff
@@ -298,14 +298,52 @@ TestSpec testResult(
             return testResult(expression, sqlExpression, result, dataType, dataType);
         }
 
+        TestSpec testResult(TableTestSpecColumn... tableTestSpecColumns) {
+            int cols = tableTestSpecColumns.length;
```

Comment:
nit: we don't enforce using final in Flink but if you check the core, you will see that most committers use final extensively to indicate immutable/mutability in code. I would vote for having a conistent coding style within our team. At least we should adapt to the coding style of the class. When I read this and the following lines, it looks to me as if we would modify the reference in the for loop again. Because all variables above are marked `final`.